### DISCUSSION TITLE
Calculate "index" on insert

### DISCUSSION THREAD

**Discussion by taylorhlabs** [prisma/prisma#5649] [Q&A] [answered]
I have a "Boards" table and a "BoardSections" table. The BoardSections table stores Sections and the order of each Section within its parent Board. When I insert a new Section, is it possible to calculate the index for the new Section? For instance, if I am adding a new Section to Board A and Board A already has 4 sections, I want the new Section to have index 5 when I insert it.

Preferably, I would be able to get the "max index"  based on the existing Section with highest index in Board A and increment it by 1. As another option, I suppose I could get the count of Sections in Board A and use the count as new index since my index starts at 0. Grateful for any solution or suggestions. Thanks!

**Comment by ryands17** [comment, marked as answer]
@taylorhlabs 👋 

One way. to do this would be find the `max` index in `Section` using `prisma.section.aggregate` and then add 1 to your new insert.

**Reply by taylorhlabs** [reply]
Thanks for the quick reply @ryands17 🙏 
Is it possible to use `aggregate` as part of the insert statement? Or do I have to do this as 2 separate calls and 2 trips to the database?

**Reply by ryands17** [reply]
I don't have any idea apart from a [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) to do this in a single call.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
